### PR TITLE
[Debt] Refactor storybook viewports

### LINKED ISSUE #5589: ♻️ Refactor storybook viewports
## ♻️ Refactor

Our use of storybook viewports relies on values that do not fit our needs and includes more viewports than we actually use.

## 🕵️ Details
We import a bunch of options we don't actually use.
### Options
https://github.com/GCTC-NTGC/gc-digital-talent/blob/01ee71da802749bf379ba24c9767b9a460851f36/frontend/common/.storybook/preview.js#L25-L26
https://github.com/GCTC-NTGC/gc-digital-talent/blob/01ee71da802749bf379ba24c9767b9a460851f36/apps/web/.storybook/preview.js#L34-L35

We only use a few that don't actually fit what we want, including the "modern" iPhone X (release in 2017).
### Screenshot
Screen Shot 2023-02-02 at 12 55 43

## 🙋‍♀️ Proposed Solution

Use [custom viewport values](https://storybook.js.org/docs/react/essentials/viewport#add-new-devices) so that we only use what we need. Consult with @GCTC-NTGC/designers for actual values.

## ✅ Acceptance Criteria

- [ ] storybook viewports work as needed

### PR DESCRIPTION
**Pull request by esizer**
🤖 Resolves #5589 

## 👋 Introduction

Updates our storybook viewports to use custom ones that align with the viewports configured.

## 🕵️ Details

This uses the values defined in the hydrogen config. That made the most sense to me but I'm open to suggestions on what ones we should include outside of that, if any.

## 🧪 Testing

Assist reviewers with steps they can take to test that the PR does what it says it does.

1. Run storybook `npm run storybook:web`
2. Confirm the viewports dropdown has values matching hydrogen config
3. Confirm chromatic snapshots exist with the proper viewports

### CODE REVIEW

**Review by mnigh — COMMENTED** [review]
Code looks great, nice work. I will defer to one of the designers to approve the visual changes, though.

File: packages/storybook-helpers/viewports.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+export type Viewport = (typeof VIEWPORT)[keyof typeof VIEWPORT];
+
+type Dimension = {
+  width: number;
+  height: number;
+};
+
+type DimensionTuple = [number, number]; // [width, height];
+
+export const DIMENSIONS: Record<Viewport, Dimension> = {
```
Comment (petertgiles) [inline packages/storybook-helpers/viewports.ts:19]: Could you leave a comment here and in the hydrogen config to help us remember to keep them in sync?

**Review by petertgiles — APPROVED** [review]
Seems to work well.  Great idea to align with hydrogen!  I left a few small things but looks good to me.

File: apps/web/.storybook/preview.js [hunk 2]
```diff
@@ -54,10 +51,7 @@ export const parameters = {
   },
   viewport: {
     // for possible values: https://github.com/storybookjs/storybook/blob/master/addons/viewport/src/defaults.ts
-    viewports: {
-      ...INITIAL_VIEWPORTS,
-      ...MINIMAL_VIEWPORTS,
-    },
+    viewports: VIEWPORTS,
```
[the remark this answers is not in the document]
Comment (esizer) [reply apps/web/.storybook/preview.js:53]: good eye! Removed in [remove old comment](https://github.com/GCTC-NTGC/gc-digital-talent/pull/8478/commits/c46c5f2f1c487811da2b67ca36ab1f94228cd610)

File: packages/storybook-helpers/viewports.ts (hunk 1 above)
Comment (esizer) [reply packages/storybook-helpers/viewports.ts:25]: Hopefully this makes sense 😅 

[add comment for width dimensions](https://github.com/GCTC-NTGC/gc-digital-talent/pull/8478/commits/3b1666fe8b56955ff7bfd5ed6c5f51195f8cb4e6)

**Pull request merged**